exp/internal/controllers/machinepool_controller_test.go

```diff
@@ -426,6 +519,11 @@ func TestReconcileMachinePoolRequest(t *testing.T) {
 			r := &MachinePoolReconciler{
 				Client:    clientFake,
 				APIReader: clientFake,
+				Tracker:   remote.NewTestClusterCacheTracker(ctrl.LoggerFrom(context.TODO()), clientFake, clientFake, clientFake.Scheme(), client.ObjectKey{Name: testCluster.Name, Namespace: testCluster.Namespace}),
+			}
+			// Set the tracker to nil for nodeDeleteTimeout cases
+			if tc.machinePool.Name == "try-delete-forever" || tc.machinePool.Name == "deleted-timeoutNode" {
```

Comment:
if checking arbitrary strings here is error prone and non starter, if needed the tracker needs to be driven by a field dedicated field in the test case. But also why do you need to do this?

Reply:
particularly valuable since we are moving this form experimental https://github.com/kubernetes-sigs/cluster-api/pull/10751

Reply:
I want the unit test to raise error by setting the tracker to nil, so when it deletes node as we expect it raise error. Otherwise we don't really know if it passes the timeout or just delete it.